src/common/utils/time.hpp

```diff
@@ -29,6 +35,22 @@ namespace utils
         {
             return std::chrono::steady_clock::now();
         }
+
+        // Returnds the current timestamp counter value. RDTSC on x86/x64, or just time since epoch for ARM
+        /// TODO: find better solution for ARM and Figure out better CPU base frequency heuristics
+        virtual uint64_t timestamp_counter()
+        {
+#if defined(_MSC_VER)
+#if defined(_M_X64) || defined(_M_AMD64) || defined(_M_IX86)
+            return __rdtsc(); // 64-bit with MSVC intrinsic
+#endif
+
+#elif defined(__x86_64__) || defined(__i386__) || defined(__amd64__) // If we are using clang or gcc
+            return __rdtsc(); // 64-bit with clang/gcc intrinsic
+#endif
+            int64_t count = std::chrono::high_resolution_clock::now().time_since_epoch().count();
```

Comment:
```suggestion
            const auto count = std::chrono::high_resolution_clock::now().time_since_epoch().count();
```

Reply:
Alright will do, I initially did use auto but thought it was supposed to be the other way around